includes/overlords/contribs.php

```diff
@@ -175,6 +207,43 @@ public static function display_contribs($mode, $id, $branch = false, $sort = fal
 			break;
 
 			case 'category' :
+				// Get a list of hidden categories, so that if a category has been set to invisible we won't show
+				// the contribs in the parent category listing
+				$all_subcategory_ids = array();
+
+				foreach ($hidden_categories_ids as $hidden_category_id)
+				{
+					$all_subcategory_ids[] = (int) $hidden_category_id;
+					$children_ids = array_keys($cache->get_category_children($hidden_category_id));
+
+					foreach ($children_ids as $child_id)
+					{
+						// Get a list of the children, it's the only way we can find out if we are in the
+						// hidden category explicitly in the next step
+						if (!in_array((int) $child_id, $all_subcategory_ids))
+						{
+							$all_subcategory_ids[] = (int) $child_id;
+						}
+					}
+				}
+
+				// We'll only remove categories if these are not equal. If they ARE equal then it means we are currently
+				// inside the hidden category (as the other ids will be the subcategories)
+				asort($all_subcategory_ids);
+				asort($id);
```

Comment:
Please fix this function's docblock which says `$id` is an integer. Obviously it can be an array of ids.

Reply:
Updated